File: odds_ends/trading-mono/api-server/src/main/java/com/revy/api_server/domain/common/BaseEntity.java
package com.revy.api_server.domain.common;

import jakarta.persistence.Column;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import lombok.ToString;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;


@Getter
@MappedSuperclass
@ToString
@EntityListeners(AuditingEntityListener.class)
public abstract class BaseEntity<T> extends BaseTimeField {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name="id")
    protected T id;
}

File: practice_kotlin/kotlin_practice_1/src/main/kotlin/com/revy/petclinic/model/BaseEntity.kt
package com.revy.petclinic.model

import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.MappedSuperclass
import java.io.Serializable

@MappedSuperclass
open class BaseEntity : Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    open var id: Long? = null;

    val isNew: Boolean
        get() = this.id != null
}